platform/os_windows.go

```diff
@@ -189,6 +206,170 @@ func SetSdnRemoteArpMacAddress() error {
+	}
+
+	if adapterNameWithVLANTag != "" {
+		//adapter is version 4 and up since adapter's advance property consists of reg key : PriorityVLANTag
+		//set reg value for Priorityvlantag of adapater to 3 if not set already
+		success, err := checkMellanoxPriorityVLANTagValueForV4(adapterName, regValueForVLANTag)
+		if err != nil {
+			log.Errorf("Error while checking registery value for PriorityVLANTag for adapater, continuing to reset the value: %v", err)
+		} else if success {
+			log.Printf("Mellanox PriorityVLANTag is already set to %v, skipping reset", regValueForVLANTag)
+			return nil
+		}
+
+		cmd = fmt.Sprintf(
+			"Set-NetAdapterAdvancedProperty -Name \"%s\" -RegistryKeyword \"%s\" -RegistryValue %d",
+			adapterName, priorityVLANTagIdentifier, regValueForVLANTag)
+		_, err = ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while setting up registery value for PriorityVLANTag for adapater: %w", err)
+		}
+
+		//check if value is set or not
+		success, err = checkMellanoxPriorityVLANTagValueForV4(adapterName, regValueForVLANTag)
+		if err != nil {
+			return fmt.Errorf("error while checking registery value for PriorityVLANTag for adapater: %w", err)
+		} else if !success {
+			return fmt.Errorf("failure while setting Mellanox Network Adapter: %s with %s property value to %d", adapterName, priorityVLANTagIdentifier, regValueForVLANTag)
+		}
+		log.Printf("Successfully set Mellanox Network Adapter: %s with %s property value as %d", adapterName, priorityVLANTagIdentifier, regValueForVLANTag)
+	} else {
+		//Adapter is version 3 or less as PriorityVLANTag was not found in advanced properties of mellanox adpater
+		log.Printf("Searching through CIM instances for Network devices with %s in the name", mellanoxSearchString)
+		cmd = fmt.Sprintf("Get-CimInstance -Namespace root/cimv2 -ClassName Win32_PNPEntity | Where-Object PNPClass -EQ \"Net\" | Where-Object { $_.Name -like \"%s\" } | Select-Object -ExpandProperty DeviceID", mellanoxSearchString)
+		deviceid, err := ExecutePowershellCommand(cmd)
+
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to get device id of %s: %w", adapterName, err)
+		}
+		if deviceid == "" {
+			return fmt.Errorf("no network device found with %s in description", mellanoxSearchString)
+		}
+
+		log.Printf("Device ID found and Getting PNP device properites for %s", deviceid)
+		cmd = fmt.Sprintf("Get-PnpDeviceProperty -InstanceId \"%s\" | Where-Object KeyName -EQ \"DEVPKEY_Device_Driver\" | Select-Object -ExpandProperty Data", deviceid)
+		registryKeySuffix, err := ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to get registery suffix of device id %s: %w", deviceid, err)
+		}
+
+		registryKeyFullPath := registryKeyPrefix + registryKeySuffix
+		success, err := checkMellanoxPriorityVLANTagValueForV3(registryKeyFullPath, adapterName, regValueForVLANTag)
```

Comment:
if this function is returning an error, is a `success` bool still necessary?

Reply:
Yes because the bool denotes if the priority vlan tag value is set to the given expected value or not, whereas the error denotes the case when we were not able to read the value.